pkg/channel/distributed/dispatcher/dispatcher/handler.go

```diff
@@ -153,6 +155,26 @@ func (h *Handler) consumeMessage(context context.Context, consumerMessage *saram
 	defer span.End()
 
 	// Dispatch The Message With Configured Retries & Return Any Errors
-	_, dispatchError := h.MessageDispatcher.DispatchMessageWithRetries(ctx, message, nil, destinationURL, replyURL, deadLetterURL, retryConfig)
+	info, dispatchError := h.MessageDispatcher.DispatchMessageWithRetries(ctx, message, nil, destinationURL, replyURL, deadLetterURL, retryConfig)
+
+	if debug {
+		// Checked Logging Level First To Avoid Calling formatDispatcherExecutionInfo In Production
+		h.Logger.Debug("Dispatcher Response: " + formatDispatcherExecutionInfo(info))
+	}
 	return dispatchError
 }
+
+func formatDispatcherExecutionInfo(info *channel.DispatchExecutionInfo) string {
+	if info == nil {
+		return "<nil>"
+	}
+	// The body for Dispatcher responses is usually empty, so don't bother printing it in those cases
+	if len(info.ResponseBody) == 0 {
+		return fmt.Sprintf("%v (%v)", info.ResponseCode, info.Time)
+	}
+	truncate := len(info.ResponseBody)
+	if truncate > 500 {
+		truncate = 500 // Responses aren't typically large, but don't dump a massive response body
+	}
+	return fmt.Sprintf("%v (%v): %v", info.ResponseCode, info.Time, string(info.ResponseBody[:truncate]))
+}
```

Comment:
seems fine - only wondering if it's worth trying to log in a more zap/json format for searching? - maybe a function that takes the execution info and makes a clone with truncated body and then use `zap.Any()` on that?  thoughts?

Reply:
I like this idea; I made a little experiment with a custom zap field (which is similar in concept) and it looks a lot better.  I'll clean it up and commit it